### PR TITLE
Nancy's Changes

### PR DESCRIPTION
**Pull request by jacobscarfmerrell**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 build
+package.json
```
Comment (jacobscarfmerrell) [inline .gitignore:2]: take this guy out

File: src/components/GamePage/index.js [hunk 2]
```diff
@@ -5,17 +5,16 @@ import Alert from 'react-bootstrap/Alert';
...
-
 import MyBoard from '../MyBoard';
 import YourBoard from '../YourBoard';
 
+
 let socket;
 
 function GamePage({ location }) {
-    const N = 6;
+    const MaxTurnLength = 6;
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:17]: `MAX_TURN_LENGTH` is conventional for constants

File: src/components/GamePage/index.js [hunk 3]
```diff
@@ -25,7 +24,7 @@ function GamePage({ location }) {
     const [myGuesses, setMyGuesses] = useState([]);
     const [yourGuesses, setYourGuesses] = useState([]);
     const [messages, setMessages] = useState([]);
-
+    const totalTurns = 11;
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:27]: `TOTAL_TURNS`

File: src/components/GamePage/index.js [hunk 4]
```diff
@@ -5,17 +5,16 @@ import Alert from 'react-bootstrap/Alert';
 import Container from 'react-bootstrap/Container';
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:5]: drop the first eslint-disable and remove any unused variables

Comment (jacobscarfmerrell) [reply src/components/GamePage/index.js:5]: you can also do PropTypes for this file if you're up to the challenge

File: src/components/GamePage/index.js [hunk 5]
```diff
@@ -35,49 +34,40 @@ function GamePage({ location }) {
     }, [ENDPOINT, location.search]);
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:34]: delete this useEffect

File: src/components/GamePage/index.js [hunk 6]
```diff
@@ -35,49 +34,40 @@ function GamePage({ location }) {
...
-                    </Row>
-                    <Row>
-                        <Form>
-                            <Form.Group controlId="formGuess">
-                                <Form.Control required type="text" onKeyPress={sendGuess} onChange={e => setGuess(e.target.value)} value={guess} />
-                            </Form.Group>
-                        </Form>
-                    </Row>
+                        {messages.map((m, i) => <Alert key={i} variant={m.variant}>{m.user} - {m.text}</Alert>)} 
+                    </Row>
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:70]: we can drop messages and lastGame

File: src/components/GamePage/index.js [hunk 7]
```diff
@@ -5,17 +5,16 @@ import Alert from 'react-bootstrap/Alert';
 import Container from 'react-bootstrap/Container';
 import Row from 'react-bootstrap/Row';
 import Col from 'react-bootstrap/Col';
-import Form from 'react-bootstrap/Form';
-
+import data from '../../data/1_board.json';
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:8]: drop data from here, import only in My/YourBoard

File: src/components/MyBoard/index.js [hunk 8]
```diff
@@ -1,24 +1,13 @@
...
-                <Body N={N} />
-            </Table>
-        </Container>
+import TurnTable from '../TurnTable';
+import data from '../../data/2_board.json';
+import './styles.css';
+
+function MyBoard() {
+    return (<div>
+        <TurnTable header={'My'} turnRecords={data.board.myturns}/>
```
Comment (jacobscarfmerrell) [inline src/components/MyBoard/index.js:8]: try replacing `header={'My'}` with `isMine`
`<TurnTable isMine turnRecords={...} />`

then you're in TurnTable
`const header = (props.isMine) ? 'My Board' : 'Your Board'`

File: src/components/TurnTable/Turn.js [hunk 9]
```diff
@@ -1,18 +1,22 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { getOffset } from '../../utilities';
+import '../MyBoard/styles.css';
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/Turn.js:4]: I would prefer to duplicate the styles file and have it at './styles.css'

File: src/components/TurnTable/Turn.js [hunk 10]
```diff
@@ -1,18 +1,22 @@
...
-function Turn({ guess, score }) {
+function Turn({ guess, score, index }) {
+    const offset = getOffset(index);
+    let currentTurn = index + 1;
     return (
         <tr>
-            <td>{guess}</td>
-            <td>{score}</td>
+            <td className={'guess-row'}>{currentTurn}. <span className={'offset'}>{offset}</span>{guess}</td>
+            <td className={'score-row'}>{score}</td>
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/Turn.js:12]: you can remove the brackets on these strings

File: src/components/TurnTable/index.js [hunk 11]
```diff
@@ -1,25 +1,32 @@
 import React from 'react';
+import Col from 'react-bootstrap/Col';
+import Row from 'react-bootstrap/Row';
 import PropTypes from 'prop-types';
+import '../MyBoard/styles.css';
 
 import Turn from './Turn';
 
-function TurnTable({turnRecords}) {
+function TurnTable({ header, turnRecords }) {
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/index.js:9]: you'll also have to change header to isMine here and in PropTypes

File: src/components/TurnTable/index.js [hunk 12]
```diff
@@ -1,25 +1,32 @@
...
+    const turns = turnRecords.length;
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/index.js:10]: turnLength

File: src/components/TurnTable/index.js [hunk 13]
```diff
@@ -28,6 +35,7 @@ TurnTable.propTypes = {
         PropTypes.shape({
             guess: PropTypes.string.isRequired,
             score: PropTypes.number.isRequired,
+            header: PropTypes.string,
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/index.js:38]: leave this as optional when you make it `isMine`

File: src/components/YourBoard/index.js [hunk 14]
```diff
@@ -1,25 +1,15 @@
...
+function YourBoard() {
     return (
-        <Container>
-            <Table striped bordered hover size="sm">
-                <Header N={N} />
-                <Body N={N} />
-            </Table>
-        </Container>
+        <Col sm={6}>
+            <TurnTable header={'Your'} turnRecords={data.board.yourturns} />
```
Comment (jacobscarfmerrell) [inline src/components/YourBoard/index.js:9]: you can drop header

File: src/components/TurnTable/index.js (hunk 13 above)
Comment (jacobscarfmerrell) [reply src/components/TurnTable/index.js:38]: you may need a defaultProps
```
TurnTable.defaultProps = {
    isMine: false,
}
```

File: src/utilities.js [hunk 15]
```diff
@@ -14,21 +12,37 @@ function getScore({ goal, guess, t, N }) {
...
+        case 5: return { currentTurnLength: MaxTurnLength - 4, offset: 4 };
+        case 6: return { currentTurnLength: MaxTurnLength - 3, offset: 3 };
+        case 7: return { currentTurnLength: MaxTurnLength - 3, offset: 2 };
+        case 8: return { currentTurnLength: MaxTurnLength - 2, offset: 0 };
+        case 9: return { currentTurnLength: MaxTurnLength - 1, offset: 0 };
+        case 10: return { currentTurnLength: MaxTurnLength - 1, offset: 2 };
+        case 11: return { currentTurnLength: MaxTurnLength, offset: 0 };
+        default: throw new Error('Invalid turn');
+    }
+}
```
Comment (jacobscarfmerrell) [inline src/utilities.js:30]: delete anything in this file you aren't currently using

File: src/components/TurnTable/index.js (hunk 12 above)
Comment (ngalter) [reply src/components/TurnTable/index.js:10]: This variable (turnLength) is gone for now.  Will most likely come back later when we add back functionality.

File: src/components/GamePage/index.js [hunk 16]
```diff
@@ -1,85 +1,37 @@
...
+                            <MyBoard />
+                        </div>
+                    </Col>
+                    <Col md={6}>
+                        <div className="d-flex justify-content-center">    
+                            <YourBoard />
+                        </div> 
+                    </Col>
+                </Row>
+            </Col>
```
Comment (jacobscarfmerrell) [inline src/components/GamePage/index.js:34]: Yes! Now we can see much better what's going on :)

File: src/components/TurnTable/index.js [hunk 17]
```diff
@@ -1,25 +1,31 @@
...
+import Col from 'react-bootstrap/Col';
+import Row from 'react-bootstrap/Row';
 import PropTypes from 'prop-types';
+import './styles.css';
 
 import Turn from './Turn';
 
-function TurnTable({turnRecords}) {
+function TurnTable({ isMine, turnRecords }) {
+    const header = isMine? 'My': 'Your' ;
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/index.js:10]: `:D`

File: src/components/TurnTable/index.js [hunk 18]
```diff
@@ -28,8 +34,13 @@ TurnTable.propTypes = {
         PropTypes.shape({
             guess: PropTypes.string.isRequired,
             score: PropTypes.number.isRequired,
+            isMine: PropTypes.bool,
```
Comment (jacobscarfmerrell) [inline src/components/TurnTable/index.js:37]: you've officially learned prop types :smile:

File: src/data/2_board.json [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+            {
+                "guess": "bbbbb",
+                "score": 3000
+            },
+            {
+                "guess": "cccccc",
+                "score": 3100
+            }
+        ],
+        "blueword": "HIDDEN"
```
Comment (jacobscarfmerrell) [inline src/data/2_board.json:95]: I like these name changes. I think they should be camel case per JSON conventions, though, e.g. `blueWord`

**Comment by jacobscarfmerrell** [comment]
@ngalter we can also do a rebase / squash lesson sometime today if you want to clean up the commits a bit. I'm fine either way!

**Comment by ngalter** [comment]
I'm available.  I don't always see my emails come in.  more likely to get a
text (609)915-3179 or a slack notice.  Let me know what you want to do.

Nancy

**Pull request merged**